tests/src/test/java/com/maxio/advancedbilling/controllers/reasoncodes/ReasonCodesControllerUpdateTest.java

```diff
@@ -0,0 +1,104 @@
+        ))).getReasonCode();
+
+        TimeUnit.SECONDS.sleep(1);
+
+        // when
+        ReasonCode updateResponse = REASON_CODES_CONTROLLER.updateReasonCode(createResponse.getId(), new UpdateReasonCodeRequest(
+                        new UpdateReasonCode("NEW_CODE", "New desc", 2)))
+                .getReasonCode();
+
+        // then
+        assertThat(updateResponse.getId()).isEqualTo(createResponse.getId());
+        assertThat(updateResponse.getSiteId()).isEqualTo(TestFixtures.SITE_ID);
+        assertThat(updateResponse.getCode()).isEqualTo("NEW_CODE");
+        assertThat(updateResponse.getPosition()).isEqualTo(2);
+        assertThat(updateResponse.getDescription()).isEqualTo("New desc");
+        assertThat(updateResponse.getCreatedAt())
+                .isNotNull()
+                .isBefore(ZonedDateTime.now());
+        assertThat(updateResponse.getUpdatedAt())
+                .isAfterOrEqualTo(updateResponse.getCreatedAt());
+
+        ReasonCode readResponse = REASON_CODES_CONTROLLER.readReasonCode(createResponse.getId()).getReasonCode();
+        assertThat(updateResponse)
+                .usingRecursiveComparison()
+                .isEqualTo(readResponse);
+    }
+
+    @Test
+    void shouldUpdateNonNullProperty() throws IOException, ApiException {
+        // given
+        ReasonCode createResponse = REASON_CODES_CONTROLLER.createReasonCode(new CreateReasonCodeRequest(new CreateReasonCode(
+                "NOT_INTERESTED", "I'm not interested in this product.", 1
+        ))).getReasonCode();
+
+        // when - then
+        ReasonCode response = REASON_CODES_CONTROLLER.updateReasonCode(createResponse.getId(), new UpdateReasonCodeRequest(
+                new UpdateReasonCode(null, "new desc", null)))
+                .getReasonCode();
+
+        assertThat(response.getCode()).isEqualTo("NOT_INTERESTED");
+        assertThat(response.getDescription()).isEqualTo("new desc");
+        assertThat(response.getPosition()).isEqualTo(1);
+    }
+
+    @Test
+    void shouldThrowExceptionIfCodeDoesNotExist() {
+        // when - then
+        assertNotFound(() -> REASON_CODES_CONTROLLER.updateReasonCode(1,
+                new UpdateReasonCodeRequest(new UpdateReasonCode("code", "desc", 1)))
+        );
+    }
+
+    @Test
+    void shouldThrowExceptionOnInvalidCredentials() throws IOException, ApiException {
+        // given
+        ReasonCode createResponse = REASON_CODES_CONTROLLER.createReasonCode(new CreateReasonCodeRequest(new CreateReasonCode(
+                "NOT_INTERESTED", "I'm not interested in this product.", 1
+        ))).getReasonCode();
+
+        // when - then
+        assertUnauthorized(() -> TestClient.createInvalidCredentialsClient().getReasonCodesController()
+                .updateReasonCode(createResponse.getId(),
+                        new UpdateReasonCodeRequest(new UpdateReasonCode("code", "desc", 1)))
+        );
+    }
```

Comment:
ive also got following error
```
HTTP Response Not OK. Status code: 422. Response: '{errors:[Code: This code is already in use.]}'
```
when ive created 2 reason codes and then tried updating code of one with code of another one

Reply:
Added error model